### ISSUE TITLE
Kotlin code style consistency with other codebases

### ISSUE THREAD

**Issue by tomholub** [FlowCrypt/flowcrypt-android#1122]
@DenBond7 this is an interesting thing - after a lot of discussion between me, Ivan and Julian on other repositories, we settled for the pattern that Ivan used above, and we disabled the IDE nudges to refactor it as you suggest.

The reason is that we'd like the code to be very understandable to people who only know Java and see Kotlin for the first time (which may be the case for the security reviewers for example, or for new team members).

And so instead of:

```kt
val x = if(cond) {
  1
} else {
  2
}
```

we'd write

```
val x: Int
if(cond) {
  x = 1
} else {
  x = 2
}
```

It's more code but we think it increases clarity. 

In situations when you really want succinct usage, we can use `when` which is I think better structured for implicit returns:

```kt
val x: Int
val x = when (cond) {
  true -> x,
  false -> y
}
```

I think that is clearer than if or try/catch with implicit returns.

Similarly with try/catch. Generally also trying to avoid implicit returns in long lambdas, like this:

```diff
whatever(something, lambda = {
  something()
  else()
-  thisIsReturnedButItsNotObvious()
+  return@lambda thisIsClearer()
})
```

We have never discussed code style much on Android because you were the only contributor and the code style was already sensible. However as we have several people who may contribute to Kotlin code these days, we should use relatively consistent styling across repositories.

(I'll file an issue with this comment)

_Originally posted by @tomholub in https://github.com/FlowCrypt/flowcrypt-android/pull/1120#discussion_r600301964_

**Comment by DenBond7** [comment]
understand. Now I see the reason. I agree with that style.

> we disabled the IDE nudges to refactor it as you suggest.

@IvanPizhenko Could you look if we can add it as repo settings?

**Comment by IvanPizhenko** [comment]
yes, I think we can do that. I'll do that after finishing some other things like decryption.

**Comment by tomholub** [comment]
copying more comments here so that style is in one place. We should probably write a document like `STYLE.md`

--------------

if/else statements do not have to have braces **if on one line**. But in that case it's only permitted to use for "consistent" result, eg like a ternary statement `return if (c) 1 else 2` or `val x = if (c) 1 else 2`. Cannot mix return and throw on one line or without braces.

if/else statements **MUST have braces if multiline**. I wish there was some ktlint rule for this (maybe there is?)

```ts
// good
val x = if(c) 1 else 2

// good
return if(c) 1 else 2

// bad
if (key is PGPSecretKeyRing)
      return key
    else
     throw IllegalArgumentException("Key is not a secret key")

// good
if (key is PGPSecretKeyRing) {
  return key
} else {
  throw IllegalArgumentException("Key is not a secret key")
}

// also good but depends on style? I prefer this but other people prefer the above. Not much difference
if (key is PGPSecretKeyRing) {
  return key
}
throw IllegalArgumentException("Key is not a secret key")
```

This is also good. Maybe the best - I find it succinct and clear:

```kt
// good
return (key as? PGPSecretKeyRing) ?: throw IllegalArgumentException("Key is not a secret key")

// if it doesn't fit on one line, a `?: throw` on next line is also good
return (key as? PGPSecretKeyRing)
  ?: throw IllegalArgumentException("Key is not a secret key")
```

I think everybody will have a little different idea, but I think this is a decent compromise.

**Comment by tomholub** [comment]
Other random thoughts:

 - ban StringBuilder except in performance-sensitive usecases (it's over-optimization to use it for 10 small strings, `+` is more readable)
 - we have no issue with singletons (`object Something { }`)
 - ban `Pair` and `Triple`, make a data class instead
 - enums preferred over strings when it makes sense

**Comment by DenBond7** [comment]
#584 #545

**Comment by DenBond7** [comment]
https://github.com/FlowCrypt/flowcrypt-android/issues/352

**Comment by IvanPizhenko** [comment]
That's all fragmented across multiple issues. I'd suggest to put everything into single markdown doc and add it into repo.

**Comment by DenBond7** [comment]
EOF([end of file](https://stackoverflow.com/questions/2287967/why-is-it-recommended-to-have-empty-line-in-the-end-of-a-source-file))

**Comment by DenBond7** [comment]
https://github.com/FlowCrypt/flowcrypt-android/pull/1316/files/3b01aec18eeabb0847f2c91dfe075cea639ffd0e#r661234994

**Comment by IvanPizhenko** [comment]
@DenBond7 @tomholub 
Since this may result in a lot of merge conflicts, this is probably best to be done when big tasks like Node removal are finished.

**Comment by tomholub** [comment]
I had the same thought, not a good time for this now.

Please help us look into #1317 next - thank you!
